core/src/test/scala/integration/kafka/api/PlaintextProducerSendTest.scala

```diff
@@ -55,6 +58,30 @@ class PlaintextProducerSendTest extends BaseProducerSendTest {
     sendAndVerify(producer)
   }
 
+  @Timeout(value = 30, unit = TimeUnit.SECONDS, threadMode = Timeout.ThreadMode.SEPARATE_THREAD)
```

Comment:
nit: 30 seconds for producing 10 records is too long IMO. Maybe reduce to 15 seconds? If it can't complete within 15 secs, there should be something wrong.